test/e2e/network_test.go

```diff
@@ -147,6 +153,37 @@ func testEastWestNetworking(t *testing.T) {
 	}
 }
 
+func (tc *testContext) getLogsAndCleanupDeployment(deployment *appsv1.Deployment) {
+	labelSelector := "app=" + deployment.Name
+	tc.savePodLogsAndEvents(labelSelector)
+	tc.deleteDeployment(deployment.Name)
+}
+
+// getLogs gets the logs for the specified job with the associated pod created by the job
+func (tc *testContext) getLogs(podLabelSelector string) (string, error) {
+	pods, err := tc.client.K8s.CoreV1().Pods(tc.workloadNamespace).List(context.TODO(), metav1.ListOptions{LabelSelector: podLabelSelector})
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting list of pods with labels %s", podLabelSelector)
+	}
+	if len(pods.Items) != 1 {
+		return "", fmt.Errorf("expected one pod and found %d", len(pods.Items))
+	}
```

Comment:
We will merge #307 once #457 merges and this is going to be a problem. The `GetLogs` API should also allow collecting logs using a selector i.e. the equivalent of `kubectl logs -l name=myLabel`.

Reply:
I did change this function to use a a podLabelSelector rather than the job name - I forgot to update the function doc above it.